pkg/render/node.go

```diff
@@ -1433,6 +1472,42 @@ func (c *nodeComponent) nodePodSecurityPolicy() *policyv1beta1.PodSecurityPolicy
 	return psp
 }
 
+// hostPathInitContainer creates an init container that changes the permissions on hostPath volumes
+// so that they can be written to by a non-root container.
+func (c *nodeComponent) hostPathInitContainer() corev1.Container {
+	rootUID := int64(0)
+	mounts := []corev1.VolumeMount{
+		{
+			MountPath: "/var/run",
+			Name:      "var-run",
+			ReadOnly:  false,
+		},
+		{
+			MountPath: "/var/lib",
+			Name:      "var-lib",
+			ReadOnly:  false,
+		},
+	}
+
+	return corev1.Container{
+		Name:         "hostpath-init",
+		Image:        "busybox:latest",
+		VolumeMounts: mounts,
+		SecurityContext: &corev1.SecurityContext{
+			RunAsUser: &rootUID,
+		},
+		Command: []string{"sh", "-c", "mkdir -p /var/lib/calico/ && chown -R 1000 /var/lib/calico/ && mkdir -p /var/run/calico/ && chown -R 1000 /var/run/calico/"},
+	}
+}
+
+// runAsNonPrivileged checks to ensure that all of the proper installation values are set for running
+// Calico as non privileged.
+func (c *nodeComponent) runAsNonPrivileged() bool {
+	return (c.cfg.Installation.NonPrivileged != nil && *c.cfg.Installation.NonPrivileged == operatorv1.NonPrivilegedEnabled) &&
+		!(c.cfg.Installation.CalicoNetwork != nil && *c.cfg.Installation.CalicoNetwork.BGP == operatorv1.BGPEnabled) &&
+		c.cfg.Installation.Variant == operatorv1.Calico
```

Comment:
Can we please consider breaking this up into separate if statements? As a one liner it's hard to understand. (It might compile down to the same thing as this one-liner anyways.)

Reply:
I think we need tests to validate the rendering